docs/content/getting-started.md

```diff
@@ -130,7 +130,8 @@ NODEPOOL_NAME=${CLUSTER_NAME}-work
 INSTANCE_TYPE=m5.2xlarge
 NODEPOOL_REPLICAS=2
 
-hypershift create nodepool --cluster-name $CLUSTER_NAME \
+hypershift create nodepool aws \
+  --cluster-name $CLUSTER_NAME \
```

Comment:
@enxebre is it valid for a NodePool's platform to differ from its owning HostedCluster?

Reply:
Nope, today that's not valid as a design invariant, it's validated here and it'll fail if you try to https://github.com/openshift/hypershift/pull/832/files#diff-d2048aaf4e4de5ddfc058be0d46748ec7e8ab85ad4bc9d62b97dae0a41138020R42-R44

In a way having `"platform" --cluster-name $CLUSTER_NAME` is redundant though I think being explicit provides a consistent better UX and helper output than implementing some CLUSTER_NAME inferring magic would do. Let me known if you think otherwise.

Reply:
I had a similar thought about how it seems weird to have to repeat the platform here given the system already knows the platform of `--cluster-name`... the validation you mentioned helps though. I was thinking repeating the platform here would only leave room for user error, but if it fails fast with a validation error, sounds fine to me.

Plus without subcommands, I'm not sure how we would sanely have per-platform arguments.

Thanks, I think it will be good!

Reply:
> Plus without subcommands, I'm not sure how we would sanely have per-platform arguments.

This is also floating around in my head for some time. In theory we could have a two-stage phase here, where we first determine the cluster type based on the name, and based on the detection register the right cobra commands. 

That could for instance work by first doin minmal pflag parsing where only platform independen flags are recognized. Then when we know the platform we could just "register" the corresponding sub-program.

Apart from the fact that it may a little bit over-engineered, the downside with this would still have, is that `--help` could only provide helpful flag output if one would point it against a valid cluster, which sounds wrong.

Reply:
> I don't know about over-engineered, but it's a really interesting idea. The nodepool subcommand as currently designed is only intended to be used in the context of an existing hostedcluster, so making the CLI "adaptive" to the target hostedcluster is pretty interesting and clever...

Thinking it through again, it may indeed not even be complex to do. @ironcladlou @nirarg , what do you think?

Reply:
to keep PR size and changes from existing code to a reasonable size I'd suggest we merge keeping existing approach and have a follow up PR to discuss this concretely?

Reply:
Works for me.